src/app/loadout-builder/LoadoutBuilder.tsx

```diff
@@ -80,6 +87,144 @@ interface State {
+            ),
+          };
+        }
+      }
+    }
+  }
+  return {
+    lockedMap,
+    statFilters: {
+      Mobility: { min: 0, max: 10, ignored: false },
+      Resilience: { min: 0, max: 10, ignored: false },
+      Recovery: { min: 0, max: 10, ignored: false },
+      Discipline: { min: 0, max: 10, ignored: false },
+      Intellect: { min: 0, max: 10, ignored: false },
+      Strength: { min: 0, max: 10, ignored: false },
+    },
+    lockedSeasonalMods: [],
+    lockedArmor2Mods: {
+      [ModPickerCategories.general]: [],
+      [ModPickerCategories.helmet]: [],
+      [ModPickerCategories.gauntlets]: [],
+      [ModPickerCategories.chest]: [],
+      [ModPickerCategories.leg]: [],
+      [ModPickerCategories.classitem]: [],
+      [ModPickerCategories.seasonal]: [],
+    },
+    minimumPower: 750,
+    query: '',
+    statOrder: statKeys,
+    selectedStoreId: getCurrentStore(stores)?.id,
+    assumeMasterwork: false,
+  };
+};
+
+export type LoadoutBuilderAction =
+  | { type: 'changeCharacter'; storeId: string }
+  | { type: 'statFiltersChanged'; statFilters: State['statFilters'] }
+  | { type: 'minimumPowerChanged'; minimumPower: number }
+  | { type: 'queryChanged'; query: string }
+  | { type: 'statOrderChanged'; statOrder: StatTypes[] }
+  | { type: 'lockedMapChanged'; lockedMap: LockedMap }
+  | { type: 'addItemToLockedMap'; item: LockedItemType }
+  | { type: 'removeItemFromLockedMap'; item: LockedItemType }
+  | { type: 'lockedSeasonalModsChanged'; lockedSeasonalMods: LockedModBase[] }
+  | {
+      type: 'lockedMapAndSeasonalModsChanged';
+      lockedMap: LockedMap;
+      lockedSeasonalMods: LockedModBase[];
+    }
+  | { type: 'lockedArmor2ModsChanged'; lockedArmor2Mods: LockedArmor2ModMap }
+  | { type: 'assumeMasterworkChanged'; assumeMasterwork: boolean };
+
+// TODO: Move more logic inside the reducer
+function stateReducer(state: State, action: LoadoutBuilderAction): State {
+  switch (action.type) {
+    case 'changeCharacter':
+      return {
+        ...state,
+        selectedStoreId: action.storeId,
+        lockedMap: {},
+        statFilters: {
+          Mobility: { min: 0, max: 10, ignored: false },
+          Resilience: { min: 0, max: 10, ignored: false },
+          Recovery: { min: 0, max: 10, ignored: false },
+          Discipline: { min: 0, max: 10, ignored: false },
+          Intellect: { min: 0, max: 10, ignored: false },
+          Strength: { min: 0, max: 10, ignored: false },
+        },
+        minimumPower: 0,
+      };
+    case 'statFiltersChanged':
+      return { ...state, statFilters: action.statFilters };
+    case 'minimumPowerChanged':
+      return { ...state, minimumPower: action.minimumPower };
+    case 'queryChanged':
+      return { ...state, query: action.query };
+    case 'statOrderChanged':
+      return { ...state, statOrder: action.statOrder };
+    case 'lockedMapChanged':
+      return { ...state, lockedMap: action.lockedMap };
+    case 'addItemToLockedMap': {
```

Comment:
I wanted to use immer for this but it would mean removing the Readonly type from LockedMap, which I was a bit hesitant to do.